backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -154,40 +162,165 @@ private fun InteropCallContext.convertIntegralToEnum(value: IrExpression, enumTy
     }
 }
 
-/**
- * Returns null if the given call-site is not an accessor for T.value, T : CEnumVar.
- * Otherwise, generates:
- *
- *  get() => byValue(this.reinterpret().value)
- *
- *  set(value) => this.reinterpret().value = value.value
+private fun InteropCallContext.readEnumValueFromMemory(nativePtr: IrExpression, enumType: IrType): IrExpression {
+    val enumClass = enumType.getClass()!!
+    val enumPrimitiveType = enumClass.properties.single { it.name.asString() == "value" }
+            .getter!!.returnType
+    val readMemory = readPrimitiveFromMemory(nativePtr, enumPrimitiveType)
+    return convertIntegralToEnum(readMemory, enumType)
+}
+
+private fun InteropCallContext.writeEnumValueToMemory(nativePtr: IrExpression, value: IrExpression): IrExpression {
+    val valueToWrite = convertEnumToIntegral(value)
+    return writePrimitiveToMemory(nativePtr, valueToWrite)
+}
+
+
+private fun InteropCallContext.calculateFieldPointer(receiver: IrExpression, offset: Long): IrExpression {
+    val base = builder.irCall(symbols.interopNativePointedRawPtrGetter).also {
+        it.dispatchReceiver = receiver
+    }
+    val nativePtrPlusLong = symbols.nativePtrType.getClass()!!
+            .functions.single { it.name.identifier == "plus" }
+    return with (builder) {
+        irCall(nativePtrPlusLong).also {
+            it.dispatchReceiver = base
+            it.putValueArgument(0, irLong(offset))
+        }
+    }
+}
+
+private fun InteropCallContext.readPointerFromMemory(nativePtr: IrExpression): IrExpression {
```

Comment:
Doesn't seem to read anything from memory.

Reply:
🤦‍♂️